src/saltext/vmware/utils/esxi.py

```diff
@@ -155,3 +159,164 @@ def list_hosts(service_instance):
+    """
+    host = utils_common.get_mor_by_property(service_instance, vim.HostSystem, name)
+    if host.summary.runtime.connectionState == "disconnected":
+        return host.summary.runtime.connectionState
+    task = host.DisconnectHost_Task()
+    host = utils_common.wait_for_task(task, name, "disconnect host task")
+    return host.summary.runtime.connectionState
+
+
+def reconnect_host(name, service_instance):
+    """
+    Reconnects host from vCenter instance
+
+    Returns connection state of host
+
+    name
+        Name of host.
+
+    service_instance
+        The Service Instance Object from which to obtain host.
+    """
+    host = utils_common.get_mor_by_property(service_instance, vim.HostSystem, name)
+    if host.summary.runtime.connectionState == "connected":
+        return host.summary.runtime.connectionState
+    task = host.ReconnectHost_Task()
+    ret_host = utils_common.wait_for_task(task, name, "reconnect host task")
+    return ret_host.summary.runtime.connectionState
+
+
+def move_host(name, cluster_name, service_instance):
+    """
+    Move host to a different cluster.
+
+    Returns connection state of host
+
+    name
+        Name of host.
+
+    cluster_name
+        Name of cluster to move host to.
+
+    service_instance
+        The Service Instance Object from which to obtain host.
+    """
+    host_ref = utils_common.get_mor_by_property(service_instance, vim.HostSystem, name)
+    cluster_ref = utils_common.get_mor_by_property(
+        service_instance, vim.ClusterComputeResource, cluster_name
+    )
+    if cluster_ref.parent.parent != host_ref.parent.parent.parent:
+        raise salt.exceptions.VMwareApiError("Cluster has to be in the same datacenter")
+    task = cluster_ref.MoveInto_Task([host_ref])
+    utils_common.wait_for_task(task, cluster_name, "move host task")
+    return "moved"
+
+
+def remove_host(name, service_instance):
+    """
+    Removes host from vCenter instance.
+
+    Returns connection state of host
+
+    name
+        Name of host.
+
+    service_instance
+        The Service Instance Object from which to obtain host.
+    """
+    host = utils_common.get_mor_by_property(service_instance, vim.HostSystem, name)
+    task = host.Destroy_Task()
+    utils_common.wait_for_task(task, name, "destroy host task")
+    return "removed"
+
+
+def _format_ssl_thumbprint(number):
+    """
+    Formats ssl cert number
+
+    number
+        Number to be formatted into ssl thumbprint
+    """
+    string = str(number)
+    return ":".join(a + b for a, b in zip(string[::2], string[1::2]))
+
+
+def _get_host_thumbprint(ip):
+    """
+    Returns host's ssl thumbprint.
+
+    ip
+        IP address of host.
+    """
+
+    ctx = ssl.SSLContext()
+    with socket.socket(socket.AF_INET, socket.SOCK_STREAM) as _socket:
+        _socket.settimeout(1)
+        with ctx.wrap_socket(_socket) as wrappedSocket:
+
+            try:
+                wrappedSocket.connect((ip, 443))
```

Comment:
**question** will this *always* use port 443, or is it configurable on the vCenter side?